### PR TITLE
Balancing the CEs

### PR DESCRIPTION
**Pull request by QuiverlyRivalry**
This, PR, aims to make the CEs more balanced out.

### CODE REVIEW

**Comment by QuiverlyRivalry** [comment]
Don’t merge, I’m at school. I’ll start workin few hours.

File: src/PiggyCustomEnchants/Tasks/EffectTask.php [hunk 1]
```diff
@@ -65,15 +65,15 @@ public function onRun(int $currentTick)
             if ($enchantment !== null) {
                 $effect = Effect::getEffect(Effect::STRENGTH);
                 $effect->setAmplifier($enchantment->getLevel() - 1);
-                $effect->setDuration(10);
+                $effect->setDuration(8);
```
Comment (DaPigGuy) [inline src/PiggyCustomEnchants/Tasks/EffectTask.php]: 🤦‍♂️

File: src/PiggyCustomEnchants/Tasks/EffectTask.php [hunk 2]
```diff
@@ -65,15 +65,15 @@ public function onRun(int $currentTick)
...
                 $effect->setVisible(false);
                 $player->addEffect($effect);
             }
             $enchantment = $this->plugin->getEnchantment($player->getInventory()->getBoots(), CustomEnchants::GEARS);
             if ($enchantment !== null) {
                 $effect = Effect::getEffect(Effect::SPEED);
                 $effect->setAmplifier(0);
-                $effect->setDuration(10);
+                $effect->setDuration(4);
```
Comment (DaPigGuy) [inline src/PiggyCustomEnchants/Tasks/EffectTask.php]: 🤦‍♂️ This isn't fixing... it's breaking.... The task runs every 5 ticks but the effect only lasts 4?

**Comment by DaPigGuy** [comment]
Duration is in ticks not seconds 🤦‍♂️

File: src/PiggyCustomEnchants/Tasks/EffectTask.php (hunk 1 above)
Comment (DaPigGuy) [reply src/PiggyCustomEnchants/Tasks/EffectTask.php]: 10 ticks to 8 ticks... Such a big difference.

**Comment by QuiverlyRivalry** [comment]
I just changed two things, since I was at school ;-;

File: src/PiggyCustomEnchants/Tasks/EffectTask.php (hunk 1 above)
Comment (QuiverlyRivalry) [reply src/PiggyCustomEnchants/Tasks/EffectTask.php]: try pvping ull see :P

**Comment by jrodr10** [comment]
I do agree on the strenght one but the other is just stupid lol

**Comment by QuiverlyRivalry** [comment]
Relax, I will redo it. I was using my iPad then. Also, do you have any suggestions on nerfs?

File: src/PiggyCustomEnchants/Tasks/EffectTask.php (hunk 1 above)
Comment (DaPigGuy) [reply src/PiggyCustomEnchants/Tasks/EffectTask.php]: 10 ticks is literally half a second...

**Comment by QuiverlyRivalry** [comment]
piggy lets floobit, im rdy

**Comment by PotatoeTrainYT** [comment]
do it yourself lmao lazy ass

**Comment by QuiverlyRivalry** [comment]
wtf, u special. he dmed me asking too floobit use ur own fuck boy

**Comment by PotatoeTrainYT** [comment]
lol

**Comment by QuiverlyRivalry** [comment]
image
get a life dumb shit

**Comment by QuiverlyRivalry** [comment]
I’ll fix it, and proof read it when I’m at home. Also I’d like to vote to remove the CE disarming... it’s just op too remove all your items

File: src/PiggyCustomEnchants/EventListener.php [hunk 3]
```diff
@@ -1064,10 +1064,10 @@ public function checkArmorEnchants(Entity $entity, Event $event)
                                 $this->plugin->cloakingcd[strtolower($entity->getName())] = time() + 10;
                                 $effect = Effect::getEffect(Effect::INVISIBILITY);
                                 $effect->setAmplifier(0);
-                                $effect->setDuration(60 * $enchantment->getLevel());
+                                $effect->setDuration(50 * $enchantment->getLevel());
                                 $effect->setVisible(false);
                                 $entity->addEffect($effect);
-                                $entity->sendMessage(TextFormat::DARK_GRAY . "You have become invisible!");
+                                $entity->sendMessage(TextFormat::DARK_GRAY . "You have become invisible!, for 5 seconds!");
```
Comment (DaPigGuy) [inline src/PiggyCustomEnchants/EventListener.php]: 🤦‍♂️ 50 ticks is 2 and a half seconds, not 5

Comment (DaPigGuy) [reply src/PiggyCustomEnchants/EventListener.php]: Not to mention the duration changes depending on level, while the message doesn't

Comment (QuiverlyRivalry) [reply src/PiggyCustomEnchants/EventListener.php]: read the top...

**Comment by DaPigGuy** [comment]
@QuiverlyRivalry It's only a 10% chance per level...

**Comment by DaPigGuy** [comment]
Instead of all this crap, we should be working on enchantment probablities, not just messing around with different durations and effect levels...

**Pull request closed without merging**